### DISCUSSION TITLE
Does the `managedResource` feature need refactoring?

### DISCUSSION THREAD

**Discussion by ndr-brt** [eclipse-edc/Connector#3289] [Ideas]
I had some lookup on the usage of the `managedResource` feature we have on the `TransferProcess` (`DataRequest`).

What is the purpose of this?
I'm not really sure about the initial needs for this, but the code is telling us that if a transfer process is `managedResource`:
- in case of no `StatusChecker` registered, the process will never get `COMPLETED`
- the connector will take care of provisioning the transfer resource(s) it on consumer side

this brings in some topics and questions about:
- why a "managed" transfer and a not managed completition is handled in a different way? Or better, why is this difference located in the `TransferProcessManager`? This should be handled by the specific `StatusChecker`.
- why is the `TransferProcessManager` the ones that decide if a resource needs to be provisioned or not? Should it be something that should be decided by the `provisioner` itself (and be part of the `DataAddress`)?

My proposal would be to have a `managed` property on the `DataAddress` object that by default it is at true, that when it's set to false, it stops the provisioning from happening, and to remove the `managedResource` from the `DataRequest` object.

any thoughts?

**Comment by jimmarino** [comment]
This is really old stuff when the EDC was first created to be used when an external host managed resource allocation and cleanup. I would remove all of it, particularly now that the DSP protocol can be used to handle this.

**Comment by ndr-brt** [comment]
done in https://github.com/eclipse-edc/Connector/pull/3297
